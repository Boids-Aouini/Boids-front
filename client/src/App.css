#app{
  display: grid;
  grid-template-columns: 9% 91%;

}

#content{
  position: absolute;
  margin-left: 9%;
  width: 91%;
}

@media only screen and (max-width: 780px){
  #app{
    grid-template-columns: 100%;
  }
}