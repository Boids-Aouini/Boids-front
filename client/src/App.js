import React, { Component } from 'react';
import './App.css';
import Register from './components/auth/register';
import {
  BrowserRouter as Router,
  Switch,
  Route
} from "react-router-dom";
import Login from './components/auth/login';
import NavBar from './components/navbar/navBar';
import LandingPage from './components/notLogedIn/landingPage';
import WhyBoids from './components/notLogedIn/whyBoids';
import { connect } from 'react-redux';
import Servers from './components/servers/servers';
import { retreiveServerAsLeader } from './actions/serverActions';

class App extends Component {

  componentWillMount() {
    if (!!this.props.auth.openedAcc) {
      this.props.retreiveServerAsLeader();
    }
  }

  render() {

    return (

      <Router>
        <div >
          <NavBar></NavBar>
          {!this.props.auth.openedAcc ? (<div></div>) : (<Servers />)}
          <Switch>
            <Route exact path="/">
              <LandingPage></LandingPage>
            </Route>
            <Route exact path="/whyBoids">
              <WhyBoids></WhyBoids>
            </Route>
            <Route exact path="/register">

              <Register />
            </Route>
            <Route exact path="/login">
              <Login></Login>
            </Route>
            {/* <Route exact path="/help">
                <div></div>
            </Route> */}

          </Switch>
        </div>
      </Router>
    )
  }
}
const mapPropsToState = state => ({
  auth: state.auth
})
export default connect(mapPropsToState, { retreiveServerAsLeader })(App);
