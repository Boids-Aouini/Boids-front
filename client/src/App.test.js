import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
let mockStore = configureStore();
let initialState = {}
let store = mockStore(initialState)
describe('<App />', () => {
  describe('render()', () => {
    test('render without crashing', () => {
      shallow(<App store={store} />);

    });
    test('render component', () => {
      let wrapper = shallow(<App store={store} />)
      let component = wrapper.dive();
      expect(toJson(component)).toMatchSnapshot()
    })

  })
})
