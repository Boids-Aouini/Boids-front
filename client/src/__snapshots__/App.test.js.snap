// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> render() render component 1`] = `
<App
  retreiveServerAsLeader={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
